lib/guardian/post_guardian.rb

```diff
@@ -269,7 +269,10 @@ def can_see_post?(post)
       return false
     end
     return true if is_moderator? || is_category_group_moderator?(post.topic.category)
-    return true if !post.trashed? || can_see_deleted_post?(post)
+    if (!post.trashed? || can_see_deleted_post?(post)) &&
+         (!post.hidden? || can_see_hidden_post?(post))
```

Comment:
I would just put these on separate lines tbh, I think it will make it easier to read...unless that loses part of the intent:

```ruby
return true if !post.trashed? || can_see_deleted_post?(post)
return true if !post.hidden? || can_see_hidden_post?(post)
```

If that is not correct just leave as is I guess :)

Reply:
This is the first thing I did, but it turns out this has the wrong behaviour. Specifically in the case where post is not trashed but hidden it exits early with `true`. 🙂